src/main/java/games/strategy/triplea/attachments/CanalAttachment.java

```diff
@@ -152,7 +152,7 @@ public void setExcludedUnits(final HashSet<UnitType> value) {
     m_excludedUnits = value;
   }
 
-  public HashSet<UnitType> getExcludedUnits(final GameData data) {
+  public HashSet<UnitType> getExcludedUnits() {
```

Comment:
Unfortunately, the tests verify that the property getter method return type exactly matches the associated field type. 😞  You can see this throughout all the attachment classes.  It appears to have been this way since the test was first introduced six years ago.